pkg/flow/adapter/xslttransform/adapter_test.go

```diff
@@ -256,6 +257,62 @@ func TestXSLTTransformEvents(t *testing.T) {
 	}
 }
 
+func TestXSLTTransformKSINK(t *testing.T) {
+	testCases := map[string]struct {
+		xslt           string
+		inEvent        cloudevents.Event
+		expectedEvent  cloudevents.Event
+		expectCategory string
+	}{
+		"transform ok": {
+			xslt:           tXSLT,
+			inEvent:        newCloudEvent(tXML, cloudevents.ApplicationXML),
+			expectedEvent:  newCloudEvent(tOutXML, cloudevents.ApplicationXML),
+			expectCategory: tSuccessAttribute,
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			ceClient := adaptertest.NewTestClient()
+
+			env := &envAccessor{
+				EnvConfig: adapter.EnvConfig{
+					Component: tCloudEventSource,
+				},
+				BridgeIdentifier: tBridgeID,
+			}
+
+			ctx := context.Background()
+			style, err := parseXSLT(tc.xslt)
+			assert.NoError(t, err)
+
+			replier, err := targetce.New(env.Component, logtesting.TestLogger(t),
+				targetce.ReplierWithStatefulHeaders(env.BridgeIdentifier),
+				targetce.ReplierWithPayloadPolicy(targetce.PayloadPolicy(targetce.PayloadPolicyAlways)),
+				targetce.ReplierWithStaticResponseType(v1alpha1.EventTypeXSLTTransformError))
+			assert.NoError(t, err)
+
+			a := &xsltTransformAdapter{
+				logger: logtesting.TestLogger(t),
+
+				ceClient:     ceClient,
+				xsltOverride: false,
+				defaultXSLT:  style,
+				replier:      replier,
+				sink:         "http://localhost:8080",
+			}
+
+			_, r := a.Dispatch(ctx, tc.inEvent)
+			assert.Equal(t, cloudevents.ResultACK, r)
+
+			events := ceClient.Sent()
+
+			assert.Equal(t, 1, len(events))
+			assert.Equal(t, tc.expectedEvent, events[0])
```

Comment:
We have to be careful with the usage of `assert` and `require`.
The first one should be `require`, otherwise the second one panics when `len(events` is 0.